url29ff2e53f3_nicolas3470_Settlers_of_Catan_AI_tgz-pJ8-robotGameJ8/src/soc/client/SOCFaceButton.java

```diff
@@ -0,0 +1,671 @@
+    private int pNumber;
+    private SOCGame game;
+    private SOCPlayerInterface pi;  // For callbacks (stack-trace print)
+
+    /** Null unless being used in the face chooser */
+    private FaceChooserFrame faceChooser;
+
+    /** Hilight selection border? always false if faceChooser == null. */
+    private boolean hilightBorderShown;
+
+    /** Recently shown hilight selection border? (Used in paint method to clear it away) Always false if faceChooser == null. */
+    private boolean hilightBorderWasShown;
+
+    /**
+     * Color for selection border; ignored if faceChooser == null.
+     *
+     * @see soc.client.SOCPlayerInterface#makeGhostColor(Color)
+     */
+    private Color hilightBorderColor;
+
+    /**
+     * Context menu for face icon chooser
+     */
+    private FaceButtonPopupMenu popupMenu;
+
+    /**
+     * Tracks last popup-menu time.  Avoids misinterpretation of popup-click with placement-click
+     * during initial placement: On Windows, popup-click must be caught in mouseReleased,
+     * but mousePressed is called immediately afterwards.
+     */
+    private long popupMenuSystime;
+
+    /** For popup-menu, length of time after popup to ignore further mouse-clicks.
+     *  Avoids Windows accidental left-click by popup-click. (150 ms)
+     */
+    protected static int POPUP_MENU_IGNORE_MS = 150;
+
+    /**
+     * offscreen buffer
+     */
+    private Image buffer;
+
+    /**
+     * size
+     */
+    protected Dimension ourSize;
+
+    private static synchronized void loadImages(Component c)
+    {
+        if (images == null)
+        {
+            tracker = new MediaTracker(c);
+            Toolkit tk = c.getToolkit();
+            Class clazz = c.getClass();
+        
+            images = new Image[NUM_FACES];
+            robotImages = new Image[NUM_ROBOT_FACES];
+
+            /**
+             * load the images
+             */
+            robotImages[0] = tk.getImage(clazz.getResource(IMAGEDIR + "/robot.gif"));
+            tracker.addImage(robotImages[0], 0);
+
+            for (int i = 1; i < NUM_FACES; i++)
+            {
+                images[i] = tk.getImage(clazz.getResource(IMAGEDIR + "/face" + i + ".gif"));
+                tracker.addImage(images[i], 0);
+            }
+
+            for (int i = 1; i < NUM_ROBOT_FACES; i++)
+            {
+                // Client possibly only has robot.gif.
+                // Check getResource vs null, and use MediaTracker;
+                // drawFace can check tracker.statusID vs MediaTracker.COMPLETE.
+                URL imgSrc = clazz.getResource(IMAGEDIR + "/robot" + i + ".gif");
+                if (imgSrc != null)
+                {
+                    robotImages[i] = tk.getImage(imgSrc);
+                    tracker.addImage(robotImages[i], i);
+                }
+            }
+
+            try
+            {
+                tracker.waitForID(0);
+            }
+            catch (InterruptedException e) {}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)